pkg/credentials/dockercreds/creds_test.go

```diff
@@ -16,133 +16,89 @@ package dockercreds
 import (
 	"flag"
 	"fmt"
-	"io/ioutil"
-	"os"
-	"path/filepath"
 	"testing"
 
-	"github.com/google/go-cmp/cmp"
+	"gotest.tools/assert"
+	"gotest.tools/env"
+	"gotest.tools/fs"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 	"github.com/knative/build-pipeline/pkg/credentials"
 )
 
 func TestFlagHandling(t *testing.T) {
-	credentials.VolumePath, _ = ioutil.TempDir("", "")
-	dir := credentials.VolumeName("foo")
-	if err := os.MkdirAll(dir, os.ModePerm); err != nil {
-		t.Fatalf("os.MkdirAll(%s) = %v", dir, err)
-	}
-	if err := ioutil.WriteFile(filepath.Join(dir, corev1.BasicAuthUsernameKey), []byte("bar"), 0777); err != nil {
-		t.Fatalf("ioutil.WriteFile(username) = %v", err)
-	}
-	if err := ioutil.WriteFile(filepath.Join(dir, corev1.BasicAuthPasswordKey), []byte("baz"), 0777); err != nil {
-		t.Fatalf("ioutil.WriteFile(password) = %v", err)
-	}
-
-	fs := flag.NewFlagSet("test", flag.ContinueOnError)
-	flags(fs)
-	err := fs.Parse([]string{
+	dir := fs.NewDir(t, "", fs.WithDir("foo",
+		fs.WithFile(corev1.BasicAuthUsernameKey, "bar"),
+		fs.WithFile(corev1.BasicAuthPasswordKey, "baz"),
+	))
+	defer dir.Remove()
+	credentials.VolumePath = dir.Path()
+
+	flagset := flag.NewFlagSet("test", flag.ContinueOnError)
+	flags(flagset)
+	assert.NilError(t, flagset.Parse([]string{
 		"-basic-docker=foo=https://us.gcr.io",
-	})
-	if err != nil {
-		t.Fatalf("flag.CommandLine.Parse() = %v", err)
-	}
-
-	os.Setenv("HOME", credentials.VolumePath)
-	if err := NewBuilder().Write(); err != nil {
-		t.Fatalf("Write() = %v", err)
-	}
+	}))
 
-	b, err := ioutil.ReadFile(filepath.Join(credentials.VolumePath, ".docker", "config.json"))
-	if err != nil {
-		t.Fatalf("ioutil.ReadFile(.docker/config.json) = %v", err)
-	}
+	defer env.Patch(t, "HOME", credentials.VolumePath)()
+	assert.NilError(t, NewBuilder().Write())
```

Comment:
my only request for uses of this `assert` lib is that we include an error message as well :pray: 

as this line currently is, i think we're going to end up with an error something like "expected result to be nil but was `docker credential write error`" and i think it just gets really hard to understand context with just that 

so maybe something lijke 

`assert.NilError(t, NewBuilder().Write(), "trying to write docker credentials to $HOME")`

(quick acknowledgement that the original error message here was pretty lacking :sweat_smile: )